client/src/components/EventPage/index.jsx

```diff
@@ -53,12 +108,82 @@ class EventPage extends Component {
         this.setState({
           category: data.category,
           tags: data.tags,
-          status_str: data.status_str,
         });
+
+        this.setState({
+          any_subs: data.any_subs,
+          status_str: data.status_str
+        })
+
+        if (data.is_subbed) {
+          this.setState({
+            is_subbed: true,
+            sub_icon: 'check_circle',
+            sub_label: 'Unsubscribe'
+          })
+        }
+        else {
+          this.setState({
+            is_subbed: false,
+            sub_icon: 'check_circle_outlined',
+            sub_label: 'Subscribe'
+          })
+        }
+        if (data.subscriptions) {
+          this.setState({
+            subs: data.subscriptions
+          })
+        }
+        else {
+          this.setState({
+            subs: []
+          })
+        }
       };
     });
   }
 
+  renderSubscribedUsers() {
+    if (this.state.any_subs) {
+      return (
+        <React.Fragment>
+        {this.state.subs.map((element) => {
+          if (element.avatar === null) {
```

Comment:
You duplicate much code here just for an avatar. Better to set it to some variable and just call it later.
Like this: `let avatar = element.avatar === null ? "/person.jpg" : element.avatar` and later ...`src={avatar}`...